### DISCUSSION TITLE
Overriding Min & Max Step Value - Woo Block Cart

### DISCUSSION THREAD

**Discussion by masteradhoc** [woocommerce/woocommerce#45009] [WooCommerce, Blocks, and Site Editing]
## Describe the bug
We often use this plugin (https://de.wordpress.org/plugins/product-quantity-for-woocommerce/) to override the min / max values to limit the number buyable of each product.
Also its nice to define steps f.e. 10 / 20 / 30 / 40 instead going up 1,2,3,4,5..

As we try to use as much of the new blocks as possible we tried it and the plugin stopped working.
We contacted the plugin author and received this response:

"We have an issue with Gutenberg block plugin, it seems the whole plugin is developed using Typescript, which runs after code compilation, I mean I can’t control that from my plugin because it generates minified files after the compilation, and in these files, they statically define step & minimum quantity of 1, making customization almost impossible. If you have a support channel with them directly, just ask them how to override the static min & step values in minified files."

## To reproduce
1. Install the plugin on a woocommerce site with woo blocks active
2. set a min / max value or steps for a product
3. check on detailed product page if it works (it does - doesnt use blocks)
4. check the same on cart (uses bloks - dosesnt work)

## Expected behavior
Allow the plugin author to override these values or hook into it.

**Comment by senadir** [comment]
Hey @masteradhoc!

Writing our code in typescript or js shouldn't matter much if we have the right APIs for extensibility, and in this case, we actually do!

Things you can do:
- Set the minimum quantity, using `woocommerce_store_api_product_quantity_minimum` filter.
- Set the maximum quantity, using `woocommerce_store_api_product_quantity_maximum` filter.
- Set the steps (multiple of), using `woocommerce_store_api_product_quantity_multiple_of` filter.
- Enable or disable quantity editing, using `woocommerce_store_api_product_quantity_editable` filter.
All of those filters will get the current limit, the product, and the cart item passed to them.

```php
add_filter(
  'woocommerce_store_api_product_quantity_multiple_of',
  function( $value, $product, $cart_item ) {
    $custom_product_id_to_limit = 25;
    if ( $custom_product_id_to_limit === $product->get_id() ) {
      return 10;
    }
    return $value;
  },
  10,
  3,
);
```

Please let us know if the author of `Min/Max, Default & Dropdown Quantity for WooCommerce` end up integrating their plugin with Cart/Checkout!

For now, I'm going to move this to discussions.

**Reply by masteradhoc** [reply]
@senadir thanks for the fast feedback!

**Reply by masteradhoc** [reply]
@senadir compatibility update gets released tomorrow. Then woo blocks are fully supported :) thanks!

**Comment by jongore** [comment]
I was also facing the same issue on my site. Actually I am working on the same business model these days. Where we offer to set tiered price restrictions by using WooCommerce maximum product plugin. Like if you want to put a limit that a specific customer can shop till 1000$ at a time. If he adds more products in the cart then an error notification will be shown to him. 

Our [WooCommerce discount plugin](https://woocommerce.com/products/cart-discounts-for-woocommerce/) also enables you to manage stock and availability of special editions products. Like you may add cart limits to certain products for controlling its sales and distribution. It can also help improve communication with customers by customizing messages shown when cart limits exceed in product, amount, quantity, weight, etc.
